code/modules/mob/living/simple_animal/hostile/space_dragon.dm

```diff
@@ -0,0 +1,727 @@
+	minbodytemp = 0
+	maxbodytemp = 1500
+	faction = list("carp")
+	pressure_resistance = 200
+	movement_type = FLYING | FLOATING // fly so you can move without gravity, float so no animation applies
+	/// Current time since the the last rift was activated.  If set to -1, does not increment.
+	var/riftTimer = 0
+	/// Maximum amount of time which can pass without a rift before Space Dragon despawns.
+	var/maxRiftTimer = 300
+	/// How much endlag using Wing Gust should apply.  Each use of wing gust increments this, and it decreases over time.
+	var/tiredness = 0
+	/// A multiplier to how much each use of wing gust should add to the tiredness variable.  Set to 5 if the current rift is destroyed.
+	var/tiredness_mult = 1
+	/// The distance Space Dragon's gust reaches
+	var/gust_distance = 4
+	/// The amount of tiredness to add to Space Dragon per use of gust
+	var/gust_tiredness = 30
+	/// Determines whether or not Space Dragon is in the middle of using wing gust.  If set to true, prevents him from moving and doing certain actions.
+	var/using_special = FALSE
+	/// Determines whether or not Space Dragon is currently tearing through a wall.
+	var/tearing_wall = FALSE
+	/// Whether space dragon is swallowing a body currently
+	var/is_swallowing = FALSE
+	/// A list of all of the rifts created by Space Dragon.  Used for setting them all to infinite carp spawn when Space Dragon wins, and removing them when Space Dragon dies.
+	var/list/obj/structure/carp_rift/rift_list = list()
+	/// How many rifts have been successfully charged
+	var/rifts_charged = 0
+	/// Whether or not Space Dragon has completed their objective, and thus triggered the ending sequence.
+	var/objective_complete = FALSE
+	/// The innate ability to use wing gust
+	var/datum/action/innate/gust_attack/gust
+	/// The innate ability to summon rifts
+	var/datum/action/innate/summon_rift/rift
+	/// The ability to make your sprite smaller
+	var/datum/action/small_sprite/space_dragon/small_sprite
+	/// The color of the space dragon.
+	var/chosen_color
+
+/mob/living/simple_animal/hostile/space_dragon/Initialize(mapload)
+	. = ..()
+	gust = new
+	gust.Grant(src)
+	rift = new
+	rift.Grant(src)
+	small_sprite = new
+	small_sprite.Grant(src)
+	RegisterSignal(small_sprite, COMSIG_ACTION_TRIGGER, .proc/add_dragon_overlay)
+
+/mob/living/simple_animal/hostile/space_dragon/Login()
+	. = ..()
+	if(!chosen_color)
+		dragon_name()
+		color_selection()
+
+/mob/living/simple_animal/hostile/space_dragon/Life(delta_time = SSMOBS_DT, times_fired)
+	. = ..()
+	tiredness = max(tiredness - 1, 0)
+	for(var/mob/living/consumed_mob in src)
+		if(consumed_mob.stat == DEAD)
+			continue
+		playsound(src, 'sound/effects/splat.ogg', 50, TRUE)
+		visible_message("<span class='danger'>[src] vomits up [consumed_mob]!</span>")
+		consumed_mob.forceMove(loc)
+		consumed_mob.Paralyze(50)
+	if((rifts_charged == 3 || (SSshuttle.emergency.mode == SHUTTLE_DOCKED && rifts_charged > 0)) && !objective_complete)
```

Comment:
They victory if the shuttle docks? Seems a bit cheap

Reply:
TG upstream said this was because crew was just calling shuttle instead of dealing with the rifts. Lore wise, everyone left and at least one rift is open, so realistically, the dragon wins, as the station is now the dragon's. I could add a check for if the dragon is on station z level.